anyblok_furetui/tests/test_call_method.py

```diff
@@ -0,0 +1,201 @@
+    class Test:
+
+        @furet_ui_call(is_classmethod=True)
+        def on_classmethod(cls, param=None):
+            return super(cls, Test).on_classmethod(param=param)
+
+        @furet_ui_call(is_classmethod=False)
+        def on_method(self, param=None):
+            return super(self, Test).on_method(param=param)
+
+        @furet_ui_call(request='request')
+        def with_request(cls, request, param=None):
+            return super(cls, Test).with_request(request, param=param)
+
+        @furet_ui_call(authenticated_userid='user')
+        def with_authenticated_userid(cls, user, param=None):
+            return super(cls, Test).with_authenticated_userid(
+                user, param=param)
+
+        @furet_ui_call(resource='resource')
+        def with_resource(cls, resource, param=None):
+            return super(cls, Test).with_resource(resource, param=param)
+
+
+def _with_call_method(oncore=False, onmixin=False, onmodel=False):
+
+    if oncore:
+        add_core_in_registry()
+
+    @register(Mixin)
+    class MixinTest:
+        pass
+
+    if onmixin:
+        add_mixin_in_registry()
+
+    @register(Model)
+    class Test(MixinTest):
+        code = String(primary_key=True)
+
+        def not_decorated(cls):
+            return True  # must be raised
+
+        def on_classmethod(cls, param=None):
+            return param
+
+        def on_method(self, param=None):
+            return self.id, param
+
+        def with_request(cls, request, param=None):
+            return request.anyblok.registry.db_name, param
+
+        def with_authenticated_userid(cls, user, param=None):
+            return user, param
+
+        def with_resource(cls, resource, param=None):
+            return resource, param
+
+    if onmodel:
+        add_model_in_registry()
+
+
+KWARGS = [
+    {'oncore': True},
+    {'onmixin': True},
+    {'onmodel': True},
+    {'oncore': True, 'onmixin': True, 'onmodel': True},
+]
+
+
+@pytest.fixture(scope="class", params=KWARGS)
+def registry_call_method(request, webserver, bloks_loaded):  # noqa F811
+    reset_db()
+    registry = init_registry_with_bloks(
+        ["furetui"], _with_call_method,
+        **request.param
+    )
+    registry.Pyramid.User.insert(login='test')
+    registry.Pyramid.CredentialStore.insert(
+       login='test', password='test')
+    webserver.post_json(
+        '/furet-ui/login', {'login': 'test', 'password': 'test'},
+        status=200)
+
+    def logout():
+        webserver.post_json('/furet-ui/logout', status=200)
+        registry.close()
+
+    request.addfinalizer(logout)
+    return registry
+
+
+class TestCallMethod:
+    @pytest.fixture(autouse=True)
+    def transact(self, request, registry_call_method):
+        transaction = registry_call_method.begin_nested()
+        request.addfinalizer(transaction.rollback)
+        return
+
+    def call(self, webserver, call, status=200):
+        url = '/furet-ui/resource/0/model/Model.Test/call/%s' % call
```

Comment:
I juste want to notice that, as far we only work with python > 3.6, I think we should avoid this old formatting operation style.

reference: https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting

> Note The formatting operations described here exhibit a variety of quirks that lead to a number of common errors (such as failing to display tuples and dictionaries correctly). Using the newer formatted string literals, the str.format() interface, or template strings may help avoid these errors. Each of these alternatives provides their own trade-offs and benefits of simplicity, flexibility, and/or extensibility.

```suggestion
        url = f'/furet-ui/resource/0/model/Model.Test/call/{call}'
```
or
```suggestion
        url = '/furet-ui/resource/0/model/Model.Test/call/{}'.format(call)
```